File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Multi Touch Bricklet 2.0

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMultiTouchV2 mt = new BrickletMultiTouchV2(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current touch state
		bool[] state = mt.GetTouchState();

		Console.WriteLine("Electrode 0: " + state[0]);
		Console.WriteLine("Electrode 1: " + state[1]);
		Console.WriteLine("Electrode 2: " + state[2]);
		Console.WriteLine("Electrode 3: " + state[3]);
		Console.WriteLine("Electrode 4: " + state[4]);
		Console.WriteLine("Electrode 5: " + state[5]);
		Console.WriteLine("Electrode 6: " + state[6]);
		Console.WriteLine("Electrode 7: " + state[7]);
		Console.WriteLine("Electrode 8: " + state[8]);
		Console.WriteLine("Electrode 9: " + state[9]);
		Console.WriteLine("Electrode 10: " + state[10]);
		Console.WriteLine("Electrode 11: " + state[11]);
		Console.WriteLine("Proximity: " + state[12]);

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{ip_connection::IpConnection, multi_touch_v2_bricklet::*};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Multi Touch Bricklet 2.0.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let mt = MultiTouchV2Bricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current touch state.
    let state = mt.get_touch_state().recv()?;

    println!("Electrode 0: {}", state[0]);
    println!("Electrode 1: {}", state[1]);
    println!("Electrode 2: {}", state[2]);
    println!("Electrode 3: {}", state[3]);
    println!("Electrode 4: {}", state[4]);
    println!("Electrode 5: {}", state[5]);
    println!("Electrode 6: {}", state[6]);
    println!("Electrode 7: {}", state[7]);
    println!("Electrode 8: {}", state[8]);
    println!("Electrode 9: {}", state[9]);
    println!("Electrode 10: {}", state[10]);
    println!("Electrode 11: {}", state[11]);
    println!("Proximity: {}", state[12]);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
